src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -53,6 +51,8 @@ internal abstract class AbstractLanguageServer<TRequestContext>
     /// </summary>
     private readonly TaskCompletionSource<object?> _serverExitedSource = new();
 
+    public AbstractTypeRefResolver TypeRefResolver { get; }
```

Comment:
for my edification, what'sa  Type-Ref resolver?  TypeRef means something very specific to me, which doesn't seem LSP realted :)

Reply:
This is the thing Dustin added to ensure that the request/response types for the MEF handlers can be lazily resolved only when a request comes in

(aka not storing the concrete Type in the MEF attribute which requires loading whatever dll the type came from).